src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -257,5 +250,13 @@ public void closeTool() {
         if ( hcEngine != null ) {
             hcEngine.shutdown();
         }
+
+        if ( referenceReader != null){
+            try {
```

Comment:
It seems wrong to have `HaplotypeCallerEngine` close the reader since it's passed in externally.   It should be the responsibility of the caller to close it.  In `HaplotypeCallerSpark` we currently broadcast an instance of ReferenceMultiSource which is reused between calls and should not be closed.

Reply:
Yes, but it's gross having the reader as a new field in the tool itself, and inconsistent with existing patterns. I think it's better to transfer responsibility for the reference reader to the `HaplotypeCallerEngine` at construction time, and let the `shutdown()` call there govern when it gets closed. The Spark version can always refrain from calling shutdown() until it's ready to do so, can't it?

Reply:
Actually, on further thought, I don't think the spark version will ever be able to properly clean up if we make this change, not without changing how it uses readers at least.  Since it should be calling shutdown whenever it's done with a haplotypecaller engine, but the readers persist through multiple engines.

Reply:
I changed it but I'm not sure if it's ideal.